Intermittent CI `_parse_ffi_str` failure

### Description

```
 _______________ ERROR collecting tests/test_arm_postprocess.py ________________
pyvex\tests\test_arm_postprocess.py:1: in <module>
    import pyvex
C:\Users\runneradmin\venv\lib\site-packages\pyvex\__init__.py:8: in <module>
    from . import const, expr, stmt
C:\Users\runneradmin\venv\lib\site-packages\pyvex\const.py:3: in <module>
    from .enums import VEXObject, get_enum_from_int
C:\Users\runneradmin\venv\lib\site-packages\pyvex\enums.py:3: in <module>
    from .native import ffi, pvc
C:\Users\runneradmin\venv\lib\site-packages\pyvex\native.py:74: in <module>
    pvc: Any = _find_c_lib()  # This should be properly typed, but this seems non trivial
C:\Users\runneradmin\venv\lib\site-packages\pyvex\native.py:64: in _find_c_lib
    _parse_ffi_str()
C:\Users\runneradmin\venv\lib\site-packages\pyvex\native.py:39: in _parse_ffi_str
    cache = pickle.loads(f.read())
E   EOFError: Ran out of input
```

Looks like an init race in: https://github.com/angr/pyvex/blob/390b91e533cba7443cfc759edd882313b282d79d/pyvex/native.py#L39 when running pytest via https://github.com/angr/pyvex/blob/390b91e533cba7443cfc759edd882313b282d79d/.github/workflows/windows.yml#L39.

### Steps to reproduce the bug

_No response_

### Environment

pyvex @ 390b91e533cba7443cfc759edd882313b282d79d

### Additional context

_No response_

--- a/pyvex/native.py
+++ b/pyvex/native.py
@@ -46,8 +46,11 @@ def _parse_ffi_str():
             "_declarations": ffi._parser._declarations,
             "_int_constants": ffi._parser._int_constants,
         }
-        with open(cache_location, "wb") as f:
-            f.write(pickle.dumps(cache))
+        # atomically write cache
+        with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+            temp_file.write(pickle.dumps(cache))
+            temp_file_name = temp_file.name
+        os.replace(temp_file_name, cache_location)
 
 
 def _find_c_lib():

